paddle/pybind/protobuf.cc

```diff
@@ -132,17 +141,7 @@ void BindProgramDesc(py::module &m) {
       .def("block", &ProgramDescBind::MutableBlock,
            py::return_value_policy::reference)
       .def("num_blocks", &ProgramDescBind::Size)
-      .def("serialize_to_string",
-           [](ProgramDescBind &program_desc) -> py::bytes {
-             const ProgramDesc *desc = program_desc.Proto();
-             PADDLE_ENFORCE(desc->IsInitialized(),
-                            "ProgramDesc has not been initialized.");
-             std::string res;
-             PADDLE_ENFORCE(
-                 desc->SerializeToString(&res),
-                 "Serialize ProgramDesc Error. This could be a bug of Paddle.");
-             return res;
-           })
+      .def("serialize_to_string", SerializeMessage<ProgramDescBind>)
```

Comment:
I think there is no need to explicitly specify the template type?

Reply:
It must specify the template type explicitly since the `def` method only accepts a function pointer as the argument.  A template function without template argument is not a valid function pointer.